fbw-a380x/src/systems/instruments/src/MFD/MFD.tsx

```diff
@@ -0,0 +1,269 @@
+            this.navigationStack.pop();
+            nextUri = this.navigationStack[this.navigationStack.length - 1];
+        } else {
+            console.info(`Navigate to ${uri}`);
+            this.navigationStack.push(uri);
+            nextUri = uri;
+        }
+
+        const parsedUri = this.parseUri(nextUri);
+        this.activeUri.set(parsedUri);
+
+        // Remove and destroy old header
+        while (this.activeHeaderRef.getOrDefault().firstChild) {
+            this.activeHeaderRef.getOrDefault().removeChild(this.activeHeaderRef.getOrDefault().firstChild);
+        }
+        if (this.activeHeader && this.activeHeader.instance instanceof DisplayComponent) {
+            this.activeHeader.instance.destroy();
+        }
+
+        // Remove and destroy old MFD page
+        while (this.activePageRef.getOrDefault().firstChild) {
+            this.activePageRef.getOrDefault().removeChild(this.activePageRef.getOrDefault().firstChild);
+        }
+        if (this.activePage && this.activePage.instance instanceof DisplayComponent) {
+            this.activePage.instance.destroy();
+        }
+
+        // Different systems use different navigation bars
+        switch (parsedUri.sys) {
+        case 'fms':
+            this.activeHeader = (
+                <FmsHeader
+                    bus={this.props.bus}
+                    callsign={Subject.create('FBW123')}
+                    activeFmsSource={this.activeFmsSource}
+                    activeUri={this.activeUri}
+                    navigateTo={(uri) => this.navigateTo(uri)}
+                />
+            );
+            break;
+        case 'atccom':
+            this.activeHeader = (
+                <AtccomHeader
+                    bus={this.props.bus}
+                    callsign={Subject.create('FBW123')}
+                    activeFmsSource={this.activeFmsSource}
+                    activeUri={this.activeUri}
+                    navigateTo={(uri) => this.navigateTo(uri)}
+                />
+            );
+            break;
+        case 'surv':
+            this.activeHeader = (
+                <SurvHeader
+                    bus={this.props.bus}
+                    callsign={Subject.create('FBW123')}
+                    activeFmsSource={this.activeFmsSource}
+                    activeUri={this.activeUri}
+                    navigateTo={(uri) => this.navigateTo(uri)}
+                />
+            );
+            break;
+        case 'fcubkup':
+            this.activeHeader = (
+                <FcuBkupHeader
+                    bus={this.props.bus}
+                    callsign={Subject.create('FBW123')}
+                    activeFmsSource={this.activeFmsSource}
+                    activeUri={this.activeUri}
+                    navigateTo={(uri) => this.navigateTo(uri)}
+                />
+            );
+            break;
+
+        default:
+            this.activeHeader = (
+                <FmsHeader
```

Comment:
They share a lot of common code, which is why I want to go the other direction: Define an abstract base class for all headers, and only implement the differences in the header classes. 
However, the content of the component is different enough so that I would want to keep separate classes